### DISCUSSION TITLE
请问如何在描述域中引用内部变量，例如目标文件名中加上构建时间

### DISCUSSION THREAD

**Discussion by TinyMiao** [xmake-io/xmake#3396] [Q&A] [answered]
#### 下面的生成目标的名称中变量并没有被替换，请问正确的用法是什么？
```
set_version("1.0.0", {build = "%Y%m%d-%H%M%S"})
target("main")
    add_files("./*.c")
    set_filename("${VERSION}_APP_$(VERSION_BUILD)_GIT$(GIT_COMMIT).app")
```

**Comment by waruqi** [comment, marked as answer]
版本变量仅支持在 configfiles 中替换。。https://xmake.io/#/zh-cn/manual/project_target?id=targetadd_configfiles

**Reply by waruqi** [reply]
你要修改文件名，自己在 after_build() 里面，自己走 os.cp/os.mv  去处理下

**Reply by TinyMiao** [reply]
这些既然是内部自带变量，后续是否会开发对外(lua内部直接引用)的使用接口呢

**Reply by waruqi** [reply]
> 这些既然是内部自带变量，后续是否会开发对外(lua内部直接引用)的使用接口呢

我文档里 可没说 VERSION 这些是 描述域的内置变量，它只仅用于 configfiles 的内置变量，仅仅在 configfiles/configvar 相关接口文档里提供了。。

这块不会考虑支持，实现不了。处理的时间点和机制完全不同。

描述的内置变量，看下文档，https://xmake.io/#/zh-cn/manual/builtin_variables 仅提供这些

**Reply by TinyMiao** [reply]
好的，非常感谢您的解答
